### PR TITLE
Fix/ignore conflicts

### LINKED ISSUE #74: Open PR even if backport is failing
- have a PR
- backport it
- backport is failing due to conflicts
- backport bot is now only saying "failing"

Instead it should open the PR so that users can fix the backport and merge it.
Otherwise/currently they have to do the backport manually again.

### PR DESCRIPTION
**Pull request by skjnldsv**
Okay, sooooo, erhm! Take a seat :couch_and_lamp: 

- [x] Bump very outdated deps
- [x] Fix audit
- [x] Cleanup
- [x] Force cherry-pick by ignoring conflicts
- [x] Notifiy in case of conflicts
  image
  image
- [x] Better debug messages

fix #74

cc @juliushaertl @christophwurst @tobiasKaminsky @rullzer @nickvergessen @danxuliu

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> Force cherry-pick by ignoring conflicts

I guess this means *any* conflict, not just the binary files from .gitattributes, right? This is a tad dangerous.

**Comment by skjnldsv** [comment]
> I guess this means _any_ conflict, not just the binary files from .gitattributes, right? This is a tad dangerous.

This is the default of cherry pick.
We also mention if there was conflicts.

It will still only try to backport the files in the pr, it's not like the compile bot that takes changed files, the only case possible is the target branch have less files than the original merged branch

**Comment by skjnldsv** [comment]
Done! Ready to review! Feel free to test on https://github.com/skjnldsv/backport-test

**Comment by skjnldsv** [comment]
I stopped the bot on my machine. ANyone willing to test it on my test repo ping me

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+services:
+  bot:
+    build: ./
+    restart: always
+    ports:
+      # you might want to use a reverse proxy anything to ssl encrypt the connexion
+      - 3000:3000
+    environment:
+      # your github app id
+      - APP_ID=12345
```
Comment (rullzer) [inline docker-compose.yml:11]: this file should not be commited I guess ;)

File: lib/pr.js [hunk 2]
```diff
@@ -1,79 +1,131 @@
...
+    const files = await context.github.pulls.listFiles(params)
+    return files.data
+  },
+
+  newReady: async function (context, origPRnr, origPRTitle, target, targetBranch) {
+    const params = context.repo({
+      title: '[' + target + '] ' + origPRTitle,
+      head: targetBranch,
+      base: target,
+      body: 'backport of #' + origPRnr
```
Comment (rullzer) [inline lib/pr.js:64]: can we also add here a big warning that the backport had conflicts?

File: docker-compose.yml (hunk 1 above)
Comment (skjnldsv) [reply docker-compose.yml:11]: Yes, it's an example file :)

File: lib/pr.js (hunk 2 above)
Comment (skjnldsv) [reply lib/pr.js:64]: Already in here ;)

Comment (skjnldsv) [reply lib/pr.js:64]: https://github.com/rullzer/backportbot/pull/179/files#diff-a39251dc5e498192873e7d036b61accfR103

Comment (nickvergessen) [reply lib/pr.js:64]: No, the link from you is in the master branch, but the backport branch also needs a warning

Comment (skjnldsv) [reply lib/pr.js:64]: Ah!! Yes!

Comment (skjnldsv) [reply lib/pr.js]: Done
image

**Comment by skjnldsv** [comment]
Anything else? :D

**Pull request merged**